Jake Jsonparser step1

ChocOZerO opened the pull request:
데이터들을 입력받아 카운트하는 로직 구현했습니다.
Bool, String, Int 타입만 체크합니다.

ChocOZerO committed:
[Add]InputView를 활용해서 입력값 받기

ChocOZerO committed:
[Add]TypeCounter를 이용하여 타입별로 카운트하기

ChocOZerO committed:
[Add]OutputView를 이용해 출력

godrm commented on JSONParser/JSONParser/TypeCounter.swift:
```diff
@@ -0,0 +1,35 @@
+    private(set) var intCounter : Int = 0
+    private(set) var boolCounter : Int = 0
+    
+    mutating func countTypes(items: Array<String>) {
```
입력한 JSON 문자열을 분석한 결과로 Array<String> 대신에 실제로도 String, Int, Bool 타입 값이 들어가는 배열로 만들어보세요.

ChocOZerO committed:
[Refactoring]JsonData 타입 추가해서 분석결과 개선

Array<String> -> JsonData ( Array<Any> ) 타입 변경
JSONAnalyser 추가해서 String을 분석하여 타입 별로 저장하도록 변경

ChocOZerO commented:
타입을 새로 만들어 적용했습니다.
그나저나 왜 저는 codacy가 이슈를 왜이리 많이 잡는걸까요;;;

godrm commented on JSONParser/JSONParser/main.swift:
```diff
@@ -8,3 +8,18 @@
+let inputItems = inputView.readInput()
+// 분석
+let analyser : JSONAnalyser = JSONAnalyser()
+let jsonData : JsonData = analyser.getJsonData(items: inputItems)
```
사소한 거지만 JSONAnalyser와 JsonData 같은 JSON 을 다루는 객체인데 소문자가 섞여있네요. 통일하는게 좋겠습니다.

godrm commented on JSONParser/JSONParserTests/JSONAnalyserTest.swift:
```diff
@@ -0,0 +1,39 @@
+
+    func testGetJsonData() {
+        let tester : String = "[ 10, \"jk\", 4, \"314\", 99, \"crong\", false ]"
+        let testers = tester.trimmingCharacters(in: ["[","]"]).split(separator: ",").map({$0.trimmingCharacters(in: .whitespaces)})
```
InputView 에 있는 코드랑 동일한 것 같은데 이렇게 중복된 부분은 적절하지 않습니다.
제 의견은 String 을 extension 으로 확장해서 동일한 함수로 쓸 수 있지 않을까요?

ChocOZerO committed:
[Update]JSONData, String Extension

JsonData -> JSONData 로 수정
getElements() 함수를 String에 Extension 해서 사용

ChocOZerO commented:
JSON명칭 통일, 공통으로 사용하는 부분 String의 extension 처리

godrm commented:
수고하셨습니다. codacy 지적사항도 초록불이 들어오도록 꾸준히 개선해보세요 👍
